helpers/internal.psm1

```diff
@@ -124,10 +124,135 @@ Function Convert-IntToIPv4 {
 #endregion Non-Exported Helpers
 
 #region Helper Functions
+Function Write-LogMessage {
+    Param (
+        $Message,
+        $LogFile
+    )
+    Write-Host "[$([System.DateTime]::Now)] $Message"
+    "[$([System.DateTime]::Now)] $Message" |  Out-File $LogFile -Append -Encoding utf8 -Width 2000
+}
+
+# ATC Deployment Status:
+#   NotDeployed: ATC is not in use - Test-NetStack will test RDMA based on whether RDMA is enabled on a given adapter.
+#   DeployedCorrectly: ATC has been deployed successfully in a configuration suitable for RDMA testing. Test-NetStack will test RDMA based on whether an adapter belongs to a storage intent.
+#   DeployedMarkFailed: ATC has been deployed, but was either unsuccessful or the configuration does not support RDMA. Test-NetStack will mark all requested RDMA stages as failures.
+enum ATCDeploymentStatus {
+    NotDeployed
+    DeployedCorrectly
+    DeployedMarkFailed
+}
+
+# This function will determine:
+#   1. Is ATC deployed?
+#   2. Are all intents successfully deployed?
+#   3. Has a storage intent been?
+# and return one of the 3 ATC deployment statuses to tell the Test-NetStack function how to test RDMA stages.
+Function Get-ATCDeploymentStatus {
+    param ($LogFile)
+
+    Write-LogMessage -Message "Determining if ATC is deployed in a supported configuration" -LogFile $LogFile
+    # Check for cluster since will need name for ATC commands on older builds
+    try {
+        $ClusterName = Get-Cluster -ErrorAction Stop
+    }
+    catch {
+        Write-LogMessage -Message "Cluster not found. If this is unexpected, please check for errors in cluster deployment. Otherwise, Test-NetStack will continue in standalone, non-ATC mode." -LogFile $LogFile

```

Comment:
Change messaging to reflect that we can't verify which adapters should be tested for RDMA, so we will make our best effort based off of which adapters have RDMA enabled. This does not account for if an adapter mistakenly has RDMA disabled.